### PR TITLE
Document expiration race condition

### PR DESCRIPTION
**Pull request by sudowork**
This commit documents the race condition and a possible workaround as described in #132.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -255,6 +255,39 @@ if (token.expired()) {
...
+opposed to an `expires_at` property), there can be discrepancies between the
+time the OAuth 2.0 server issues the access token and when it is received.
+These come down to factors such as network and processing latency. This can be
+worked around by preemptively refreshing the access token:
+
+```javascript
+// Provide a window of time before the actual expiration to refresh the token
+const EXPIRATION_WINDOW_IN_SECONDS = 300;
+
+const tokenObject = token.token;
```
Comment (sudowork) [inline README.md]: I maybe would have preferred if the actual `AccessToken` object was assigned to the variable `accessToken`:

```javascript
// Create the access token wrapper
const accessToken = oauth2.accessToken.create(tokenObject);

// ...

const { token } = accessToken;
const expirationTimeInSeconds = token.expires_at.getTime() / 1000;

/// ...
```

Comment (jonathansamines) [inline README.md]: Please go ahead and update the reference example :)

**Comment by sudowork** [comment]
@jonathansamines Updated the documentation.

File: README.md [hunk 2]
```diff
@@ -140,14 +140,14 @@ oauth2.authorizationCode.getToken(tokenConfig, (error, result) => {
     return console.log('Access Token Error', error.message);
   }
 
-  const token = oauth2.accessToken.create(result);
+  const accessToken = oauth2.accessToken.create(result);
```
Comment (sudowork) [inline README.md]: Updated these examples to be consistent.

File: README.md [hunk 3]
```diff
@@ -238,19 +238,19 @@ const tokenObject = {
 };
 
 // Create the access token wrapper
-const token = oauth2.accessToken.create(tokenObject);
+let accessToken = oauth2.accessToken.create(tokenObject);
```
Comment (sudowork) [inline README.md]: Fixed this from being a `const` to a `let` since it gets reassigned.

**Comment by sudowork** [comment]
@jonathansamines Any other changes you would like me to make?

**Comment by jonathansamines** [comment]
@sudowork Any at all, great work @sudowork will merge as is. Thanks!

**Pull request merged**